File: client/scalaz/src/main/scala/org/openapitools/client/api/AdamVariable.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import AdamVariable._

case class AdamVariable (
  ordinal: Option[String],
name: Option[String],
label: Option[String],
description: Option[String],
core: Option[String],
simpleDatatype: Option[String],
describedValueDomain: Option[String],
valueList: Option[List[String]],
links: Option[AdamVariableLinks])

object AdamVariable {
  import DateTimeCodecs._

  implicit val AdamVariableCodecJson: CodecJson[AdamVariable] = CodecJson.derive[AdamVariable]
  implicit val AdamVariableDecoder: EntityDecoder[AdamVariable] = jsonOf[AdamVariable]
  implicit val AdamVariableEncoder: EntityEncoder[AdamVariable] = jsonEncoderOf[AdamVariable]
}

File: src/cdisc_library_api/specs/adam_variable.clj
(ns cdisc-library-api.specs.adam-variable
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [cdisc-library-api.specs.adam-variable-links :refer :all]
            )
  (:import (java.io File)))


(def adam-variable-data
  {
   (ds/opt :ordinal) string?
   (ds/opt :name) string?
   (ds/opt :label) string?
   (ds/opt :description) string?
   (ds/opt :core) string?
   (ds/opt :simpleDatatype) string?
   (ds/opt :describedValueDomain) string?
   (ds/opt :valueList) (s/coll-of string?)
   (ds/opt :_links) adam-variable-links-spec
   })

(def adam-variable-spec
  (ds/spec
    {:name ::adam-variable
     :spec adam-variable-data}))
